include/igl/eigs.cpp

```diff
@@ -33,8 +34,8 @@ IGL_INLINE bool igl::eigs(
   assert(iB.rows() == n && "B should be match A's dims.");
   assert(iB.cols() == n && "B should be square.");
   assert(type == EIGS_TYPE_SM && "Only low frequencies are supported");
-  DerivedU U(n,k);
-  DerivedS S(k,1);
+  DerivedU U(n, 1);
```

Comment:
It would save the resize cost if `U` was just set to an nxn, `U(n,n)`, and then a block copy was returned after max_iter iterations depending on the number of pairs. I suppose the question is which is more expensive in common situations, resizing U (conservativeResize, if I recall, is not an inexpensive operation, for obvious reasons) or preallocating a large U and copying a block at the end (is `n `>> `k` always for some reasonable number of max_iter?) 

I don't mind making this change if it turns out the latter is the better approach.